exercises/verena_reimund/ex02_oo_basics/src/main.cpp

```diff
@@ -1,181 +1,159 @@
 /**
-* New type: Shapes (circle, ..) with functionality defined in class ShapeSorter
+* New type: shape (circle, ..) with functionality defined in class shapeorter
 */
-#include <math.h>    //pow
+#include <math.h>    //pow, M_PI
 #include <iostream>
 #include <vector>
 #include <string>
 #include <algorithm> //for_each
 //#include <memory>  //smart pointers
 
-
-// base class
-class print{
-public:// overloaded print functions
-   void out (std::string type, double val);
-   void out (std::string type, std::string val);
-   void out (std::string type, double val1, double val2);
-};
-
 // abstract base class
-class Shapes: public print{
-   std::string n;
-   double shape_area;
-   double shape_perimeter;
-   std::string n_sides;
+class shape{
 public:
-   Shapes(){}
-   virtual std::string type() const {return n;}
-   virtual std::string Square() const {return n_sides;}
-   virtual double perimeter () const {return shape_perimeter;}
-   virtual double area () const {return shape_area;}
-   virtual ~Shapes(){}
+    shape(){}
```

Comment:
This is exactly the way I think it should be done. 👍 

As a side-topic, when you're using c++11, I suggest using the new `override` keyword wherever possible. You can't use it for the current exercise, but you can in Mantid.